File: Java/src/main/java/QuestionBank/Solution701_800/Solution724.java
package QuestionBank.Solution701_800;

/**
 * @author windows
 */
public class Solution724 {
    public int pivotIndex(int[] nums) {
        int left = 0;
        int right = 0;
        for (int i = 1; i < nums.length; i++) {
            right += nums[i];
        }
        int i = 0;
        while (left != right && i < nums.length - 1) {
            i++;
            left += nums[i - 1];
            right -= nums[i];
        }
        if (left == right){
            return i;
        }
        return -1;
    }


    public static void main(String[] args) {
        Solution724 solution724 = new Solution724();
        int index = solution724.pivotIndex(new int[]{1, 2, 3});
        System.out.println(index);
    }
}

File: C++/src/Solution701_800/Solution724.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution724.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution724::pivotIndex(vector<int> &nums) {
    int left_sum = 0;
    int right_sum = 0;
    for (int i = 0; i < nums.size(); ++i) {
        right_sum += nums[i];
    }
    for (int i = 0; i < nums.size(); ++i) {
        right_sum -= nums[i];
        if (left_sum == right_sum){
            return i;
        }else{
            left_sum += nums[i];
        }
    }
    return -1;
}
